File: LABS/LAB_03/src/ki304/vasylkiv/lab3/PersonalInfo.java
package ki304.vasylkiv.lab3;

/**
 * The <code>PersonalInfo</code> class represents personal information about a person.
 */
public class PersonalInfo {
    private String firstName;
    private String lastName;
    private int age;

    /**
     * Default constructor that initializes fields with default values.
     */
    public PersonalInfo() {
    	firstName = "unknown";
    	lastName = "unknown";
    	age = 0;
    }

    /**
     * Constructor that takes values for first name, last name, and age.
     * 
     * @param newFirstName  New first name
     * @param newLastName   New last name
     * @param newAge        New age
     */
    public PersonalInfo(String newFirstName, String newLastName, int newAge) {
    	firstName = newFirstName;
    	lastName = newLastName;
    	age = newAge;
    }
    
    /**
     * Method is used to change the first name.
     *
     * @param newFirstName New first name
     */
    public void setFirstName(String newFirstName) {
        firstName = newFirstName;
    }

    /** 
     * Method returns first name.
     * @return String
     */
    public String getFirstName() {
        return firstName;
    }

    /**
     * Method is used to change the last name.
     *
     * @param newLastName New last name
     */
    public void setLastName(String newLastName) {
        lastName = newLastName;
    }

    /** 
     * Method returns last name.
     * @return String
     */
    public String getLastName() {
        return lastName;
    }

    /**
     * Method is used to change the age.
     *
     * @param newAge New age
     */
    public void setAge(int newAge) {
        age = newAge;
    }

    /** 
     * Method returns age.
     * @return int
     */
    public int getAge() {
        return age;
    }

    /**
     * Method is used to change the age.
     *
     * @param newAge New age
     */
    public void changeAge(int newAge) {
        setAge(newAge);
    }

    /**
     * Method is used to increase age by one year.
     */
    public void growUp() {
    	age ++;
    }

    /** 
     * Method returns full name.
     * @return String
     */
    public String getFullName() {
    	return firstName + " " + lastName;
    }
}

File: LABS/LAB_09/PersonalInfo.py
class PersonalInfo:
    
    # Constructor
    def __init__(self, first_name="unknown", last_name="unknown", age=0):
        self.__first_name = first_name
        self.__last_name = last_name
        self.__age = age

    # Method: Sets the first_name attribute of the PersonalInfo object.
    def setFirstName(self, new_first_name):
        self.__first_name = new_first_name

    # Method: Returns the first_name attribute of the PersonalInfo object.
    def getFirstName(self):
        return self.__first_name

    # Method: Sets the last_name attribute of the PersonalInfo object.
    def setLastName(self, new_last_name):
        self.__last_name = new_last_name

    # Method: Returns the last_name attribute of the PersonalInfo object.
    def getLastName(self):
        return self.__last_name

    # Method: Sets the age attribute of the PersonalInfo object.
    def setAge(self, new_age):
        self.__age = new_age

    # Method: Returns the age attribute of the PersonalInfo object.
    def getAge(self):
        return self.__age

    # Method: Changes the age attribute of the PersonalInfo object.
    def changeAge(self, new_age):
        self.setAge(new_age)

    # Method: Increases the age attribute of the PersonalInfo object by 1.
    def growUp(self):
        self.__age += 1

    # Method: Returns a formatted string with the full name of the PersonalInfo object.
    def getFullName(self):
        return f"{self.__first_name} {self.__last_name}"
